sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -87,12 +87,17 @@ int coap_parse(coap_pkt_t *pkt, uint8_t *buf, size_t len)
         pkt->token = NULL;
     }
 
+    if (pkt_pos > pkt_end) {
```

Comment:
Let's combine checks and save a few bytes. With your change to the while loop condition, we can move this check below the while loop, and remove the check currently at the end of the while loop. I'd change the debug message to something like "bad pkt len" to cover both cases. 

It also would be good to add a comment above, where testing the token length, that the validation is below the while loop.